agent/rpc/peering/service.go

```diff
@@ -586,17 +605,19 @@ func (s *Server) getExistingOrCreateNewPeerID(peerName, partition string) (strin
 
 func copyPeeringWithNewState(p *pbpeering.Peering, state pbpeering.PeeringState) *pbpeering.Peering {
 	return &pbpeering.Peering{
```

Comment:
I just realized this whole function could be:

```
copy := *p
copy.State = state
return &copy
```

That way if we add fields we dont keep needing to  add more manual copying of fields here

Reply:
ah actually this seems to make `go vet` understandably unhappy. Gotta love go vet ❤️ 

https://app.circleci.com/pipelines/github/hashicorp/consul/35170/workflows/565b2994-d7ad-4da8-8825-baee49215c59/jobs/767931